### PR TITLE
Official Ably Client Broadcaster Doc

### PR DESCRIPTION
**Pull request by sacOO7**
- Updated `broadcaster.md` as per latest `ably client`
- Marked `old AblyBroadcaster` section as `Deprecated`

### CODE REVIEW

File: broadcasting.md [hunk 1]
```diff
@@ -208,44 +208,126 @@ window.Echo = new Echo({
...
-> **Note**  
-> To learn more about compiling your application's JavaScript assets, please consult the documentation on [Vite](/docs/{{version}}/vite).
+
+> {tip} To learn more about compiling your application's JavaScript assets, please consult the documentation on [Laravel Mix](/docs/{{version}}/mix).
+
+### Additional supported features (official ably client)
+
+**1. Update token expiry (Default 3600, 1 hr)**
+- Update`ABLY_TOKEN_EXPIRY` in `.env` file. 
+- Update `ably` section under `config/broaadcasting.php` with `'token_expiry' => env('ABLY_TOKEN_EXPIRY', 3600)`
```
Comment (qsdstefan) [inline broadcasting.md]: Typo, `config/broadcasting.php`

File: broadcasting.md [hunk 2]
```diff
@@ -208,44 +208,126 @@ window.Echo = new Echo({
...
+>    wsHost: 'realtime-pusher.ably.io',
+>    wsPort: 443,
+>    disableStats: true,
+>    encrypted: true,
+>});
+>```
+>
+>Note that our Ably Echo configuration references a `MIX_ABLY_PUBLIC_KEY` environment variable. This variable's value should be your Ably public key. Your public key is the portion of your Ably key that occurs before the `:` character.
+>
+>Update php-Laravel config. - Set `ABLY_PUSHER_ADAPTER` as **true** in `.env` file. Also, make sure `ably` section under `config/broadcasting.php` is updated with `'pusher_adapter' => env('ABLY_PUSHER_ADAPTER', true)`
```
Comment (qsdstefan) [inline broadcasting.md]: Second part is not required, and isn't common practice in Laravel, as it's the default value and should be left at `false`. The purpose of the `.env` file and `env` function is to allow users to use custom values instead of the default specified in `config/broadcasting.php` or another config file.

File: broadcasting.md (hunk 1 above)
Comment (sacOO7) [reply broadcasting.md]: Updated!

File: broadcasting.md (hunk 2 above)
Comment (sacOO7) [reply broadcasting.md]: Updated!

**Review by qsdstefan — APPROVED** [review]
Nice, LGTM!

File: broadcasting.md [hunk 3]
```diff
@@ -208,44 +208,126 @@ window.Echo = new Echo({
 <a name="client-ably"></a>
 ### Ably
 
-[Laravel Echo](https://github.com/laravel/echo) is a JavaScript library that makes it painless to subscribe to channels and listen for events broadcast by your server-side broadcasting driver. You may install Echo via the NPM package manager. In this example, we will also install the `pusher-js` package.
+[Laravel Echo](https://github.com/laravel/echo) is a JavaScript library that makes it painless to subscribe to channels and listen for events broadcast by your server-side broadcasting driver.
 
-You may wonder why we would install the `pusher-js` JavaScript library even though we are using Ably to broadcast our events. Thankfully, Ably includes a Pusher compatibility mode which lets us use the Pusher protocol when listening for events in our client-side application:
+Install **official** `ably` package along with `laravel-echo` using NPM.
```
Comment (owenpearson) [inline broadcasting.md]: I think we should make this section similar to the Pusher channels section:

```suggestion
[Laravel Echo](https://github.com/laravel/echo) is a JavaScript library that makes it painless to subscribe to channels and listen for events broadcast by your server-side broadcasting driver. You may install Echo via the NPM package manager. In this example, we will also install the ably package since we will be using the Ably broadcaster:
```

File: broadcasting.md [hunk 4]
```diff
@@ -208,44 +208,126 @@ window.Echo = new Echo({
...
+
+window.Echo.connector.ably.connection.on(stateChange => {
+    if (stateChange.current === 'connected') {
+        console.log('connected to ably server');
+    }
 });
 ```
 
-Note that our Ably Echo configuration references a `VITE_ABLY_PUBLIC_KEY` environment variable. This variable's value should be your Ably public key. Your public key is the portion of your Ably key that occurs before the `:` character.
+You can set custom [clientOptions](https://docs.ably.io/client-lib-development-guide/features/#options) while creating `Echo` instance.
```
Comment (owenpearson) [inline broadcasting.md]:
```suggestion
You can set custom [clientOptions](https://ably.com/docs/api/realtime-sdk#client-options) when creating an `Echo` instance.
```

File: broadcasting.md [hunk 5]
```diff
@@ -208,44 +208,126 @@ window.Echo = new Echo({
...
 
+```
+    broadcaster: 'ably',
+    authEndpoint: 'http://www.localhost:8000/broadcasting/auth'
+    // Additional ably specific options - https://docs.ably.io/client-lib-development-guide/features/#options
```
Comment (owenpearson) [inline broadcasting.md]:
```suggestion
    // Additional ably specific options - https://ably.com/docs/api/realtime-sdk#client-options  
```

File: broadcasting.md [hunk 6]
```diff
@@ -208,44 +208,126 @@ window.Echo = new Echo({
...
 Once you have uncommented and adjusted the Echo configuration according to your needs, you may compile your application's assets:
 
 ```shell
 npm run dev
 ```
 
-> **Note**  
-> To learn more about compiling your application's JavaScript assets, please consult the documentation on [Vite](/docs/{{version}}/vite).
+
+> {tip} To learn more about compiling your application's JavaScript assets, please consult the documentation on [Laravel Mix](/docs/{{version}}/mix).
```
Comment (owenpearson) [inline broadcasting.md]: Is this change needed? Doesn't seem to be related to the changes we're making.

File: broadcasting.md [hunk 7]
```diff
@@ -208,44 +208,126 @@ window.Echo = new Echo({
...
+
+### Additional supported features (official ably client)
+
+**1. Update token expiry (Default 3600, 1 hr)**
```
Comment (owenpearson) [inline broadcasting.md]:
```suggestion
**1. Update token expiry. Default: 3600 seconds (1 hr)**
```

File: broadcasting.md [hunk 8]
```diff
@@ -208,44 +208,126 @@ window.Echo = new Echo({
...
+- Update`ABLY_TOKEN_EXPIRY` in `.env` file. 
+- Update `ably` section under `config/broadcasting.php` with `'token_expiry' => env('ABLY_TOKEN_EXPIRY', 3600)`
+
+**2. Modify channel capability**
+- Channel access can be changed as per [Channel Capabilities](https://ably.com/docs/core-features/authentication#capability-operations)
+```php
+  // file - routes/channels.php
+
+  // for private channel (array is returned as a truthy value)
```
Comment (owenpearson) [inline broadcasting.md]: The sentence "array is returned as a truthy value" is a bit confusing to me, I'm not sure what you're trying to say here.

File: broadcasting.md [hunk 9]
```diff
@@ -208,44 +208,126 @@ window.Echo = new Echo({
...
+>    encrypted: true,
+>});
+>```
+>
+>Note that our Ably Echo configuration references a `MIX_ABLY_PUBLIC_KEY` environment variable. This variable's value should be your Ably public key. Your public key is the portion of your Ably key that occurs before the `:` character.
+>
+>Update php-Laravel config. - Set `ABLY_PUSHER_ADAPTER` as **true** in `.env` file. Also, make sure `ably` section under `config/broadcasting.php` is updated with `'pusher_adapter' => env('ABLY_PUSHER_ADAPTER', false)`
+
+### Migrating from deprecated ably-pusher adapter to ably
+- Latest Ably broadcaster is fully compatible with old Ably-Pusher broadcaster.
```
Comment (owenpearson) [inline broadcasting.md]:
```suggestion
- The latest Ably broadcaster is fully compatible with the deprecated Ably-Pusher broadcaster.
```

File: broadcasting.md [hunk 10]
```diff
@@ -208,44 +208,126 @@ window.Echo = new Echo({
...
+- Only **Leaving the channel**, [Use Ably Channel Namespace](https://ably.com/docs/general/channel-rules-namespaces) conventions
```
Comment (owenpearson) [inline broadcasting.md]:
```suggestion
- The only API difference is for **leaving a channel**, you must [use Ably Channel Namespace](https://ably.com/docs/general/channel-rules-namespaces) conventions
```

**Review by owenpearson — CHANGES_REQUESTED** [review]

File: broadcasting.md (hunk 6 above)
Comment (sacOO7) [reply broadcasting.md]: Ohh yes, you are right!

File: broadcasting.md (hunk 8 above)
Comment (sacOO7) [reply broadcasting.md]: It has to do with returning either `true` or `false`. When `truthy` value is returned, channel access is given. So, it doesn't matter if the returned value is of type `array` or `object` or `boolean`, all of them are treated as `true` values. It's the same concept of `Truthy` and `Falsy` values in js -> https://www.sitepoint.com/javascript-truthy-falsy/

File: broadcasting.md (hunk 4 above)
Comment (owenpearson) [reply broadcasting.md]: This comment still stands - we should link to the API reference docs rather than the features spec

File: broadcasting.md (hunk 8 above)
Comment (owenpearson) [reply broadcasting.md]: OK yeah I get that but I still think this could be clearer - maybe it should say:
```suggestion
  // for private channel (If a falsy value is returned then channel access is not permitted for the client)
```

also FWIW empty arrays are falsy in PHP

File: broadcasting.md (hunk 4 above)
Comment (sacOO7) [reply broadcasting.md]: Actually, I feel the features spec is misleading. You can take a look at options, I think they are following `ruby` conventions for `snake_case` options.

File: broadcasting.md (hunk 8 above)
Comment (sacOO7) [reply broadcasting.md]: Updated to ->
`// for private channel (Access is allowed for truthy values and denied for falsy values)`

File: broadcasting.md (hunk 4 above)
Comment (sacOO7) [reply broadcasting.md]: Updated!

**Review by owenpearson — APPROVED** [review]
LGTM 👍

**Comment by sacOO7** [comment]
Thanks for approval, merging the PR since all comments are addressed!
